File: javaScript/two-sum.js
var twoSum = function (nums, target) {
  if (nums == null || nums.length == 1) return [];

  var compMap = {};

  for (let i = 0; i < nums.length; i++) {
    var comp = target - nums[i];
    if (compMap[comp] >= 0) {
      return [compMap[comp], i];
    }
    compMap[nums[i]] = i;
  }

  return [];
};

File: csharp/two-sum.cs
// Time: O(n)
// Space: O(n)
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var dict = new Dictionary<int, int>();
        
        for(int i = 0; i < nums.Length; i++)
        {
            if(dict.ContainsKey(nums[i])) return new int[] { i, dict[nums[i]] };
            if(!dict.ContainsKey(target - nums[i])) dict.Add(target - nums[i], i);
        }
        
        return null;
    }
}